docs/topics/yaml-rule-structure-syntax.adoc

```diff
@@ -0,0 +1,520 @@
+* `potential`
+** You need to examine the issue during the migration process, but there is not enough information to determine whether resolving the issue is mandatory for the migration to succeed. An example of such an issue is migrating a third-party proprietary type when there is no directly compatible type on the target platform.
+
+
+[id="yaml-rule-actions_{context}"]
+=== Rule Actions
+
+Rules can include 2 types of actions: _message_ and _tag_. Each rule includes one of them or both.
+
+.Message actions
+
+The message action creates an issue with a message when the rule matches. The custom data exported by providers can also be used in the message.
+
+`message: "helpful message about the issue"`
+
+_Example:_
+
+[source,terminal]
+----
+- ruleID: test-rule
+  when:
+    <CONDITION>
+  message: Test rule matched. Please resolve this migration issue.
+----
+
+Optionally, a message can include hyperlinks to external URLs that provide relevant information about the issue or a quick fix.
+
+[source,terminal]
+----
+links:
+  - url: "konveyor.io"
+    title: "Short title for the link"
+----
+
+Message can also be templated to include information about the match interpolated through custom variables on the rule.
+
+.Tag actions
+
+The tag action instructs the analyzer to generate one or more tags for the application when a match is found. Each string in the `tag` field can be a comma-separated list of tags. Optionally, you can assign categories to tags.
+
+[source,terminal]
+----
+tag:
+  - "tag1,tag2,tag3"
+  - "Category=tag4,tag5"
+----
+
+_Example_
+
+[source,terminal]
+----
+- ruleID: test-rule
+  when:
+    <CONDITION>
+  tag:
+  - Language=Golang
+  - Env=production
+  - Source Code
+----
+
+A tag can be a string or a `key=val` pair, where the key is treated as a tag category in {ProductShortName}. Any rule that has a tag action is referred to as a “tagging rule” in this document.
+
+Note that issues are not created for rules that contain only tag actions.
+
+
+[id="yaml-rule-conditions_{context}"]
+=== Rule conditions
+
+Each rule has a `when` block, which specifies a condition that needs to be met for {ProductShortName} to perform a certain action.
+
+The `when` block contains one condition, but that condition can have multiple conditions nested under it.
+
+[source,terminal]
+----
+when:
+  <condition>
+    <nested-condition>
+----
+
+{ProductShortName} supports three types of conditions: `provider`, `and` and `or`.
+
+==== Provider conditions
+
+{ProductShortName} supports multi-language source code analysis. Source code search for a specific language is enabled using the `provider` condition. This condition defines a search query for a specific language provider.  The `provider` condition also specifies which of the provider's "capabilities" to use for analyzing the code.
+
```

Comment:
"source code search" is a bit of a noun cluster. Maybe "You can search source code for a specific language by using..." Probably a lot of sentences should start with "you" but that should be a post-GA consideration.